ussomacaroon/client.go

```diff
@@ -0,0 +1,103 @@
+// Copyright 2016 Canonical Ltd.
+// Licensed under the LGPLv3, see LICENCE file for details.
+
+// Pacakge ussomacaroon provides a client that can authenticate with an
+// identity server by discharging macaroons on an Ubuntu SSO server.
+package ussomacaroon
+
+import (
+	"net/url"
+
+	"github.com/juju/httprequest"
+	errgo "gopkg.in/errgo.v1"
+	"gopkg.in/macaroon-bakery.v2-unstable/bakery"
+	"gopkg.in/macaroon-bakery.v2-unstable/httpbakery"
+	"gopkg.in/macaroon.v2-unstable"
+)
+
+// Macaroon returns a macaroon from the identity provider at the given
+// URL, which will have a third-party caveat
+// addressed to an Ubuntu SSO service.
+func Macaroon(url string) (*macaroon.Macaroon, error) {
+	client := &httprequest.Client{}
+	var resp MacaroonResponse
+	if err := client.Get(url, &resp); err != nil {
+		return nil, errgo.Notef(err, "cannot get macaroon")
+	}
+	return resp.Macaroon, nil
+}
+
+// Visitor is an httpbakery.Visitor that will login using a macaroon
```

Comment:
perhaps in addition to, but also those sort of things should be in the test files.